clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2857,29 +2860,33 @@ class SYCLKernelNameTypeVisitor
     } else {
       InnerTypeVisitor::Visit(T.getTypePtr());
     }
+    return true;
   }
 
-  void Visit(const TemplateArgument &TA) {
+  bool Visit(const TemplateArgument &TA) {
     if (TA.isNull())
-      return;
-    InnerTAVisitor::Visit(TA);
+      return false;
+    return InnerTAVisitor::Visit(TA);
   }
 
-  void VisitEnumType(const EnumType *T) {
+  bool VisitEnumType(const EnumType *T) {
     const EnumDecl *ED = T->getDecl();
     if (!ED->isScoped() && !ED->isFixed()) {
       S.Diag(KernelInvocationFuncLoc, diag::err_sycl_kernel_incorrectly_named)
           << /* Unscoped enum requires fixed underlying type */ 2;
       S.Diag(ED->getSourceRange().getBegin(), diag::note_entity_declared_at)
           << ED;
+      IsInvalid = true;
+      return isValid();
     }
+    return true;
```

Comment:
This return was added so that the compiler wouldn't throw this warning: `control reaches end of non-void function [-Wreturn-type]`. If the condition inside the "if" clause is not true, the compiler expects a return statement outside the "if" block, since this method now returns a bool.